web/src/components/new/TagCreateDialog.vue

```diff
@@ -41,18 +55,36 @@ export default {
       tagStreams: [],
       tagType: "",
       tagName: "",
+      tagColor: "",
+      tagColorPickerOpen: false,
     };
   },
   created() {
     EventBus.$on("showCreateTagDialog", this.openDialog);
   },
+  computed: {
+    // https://codepen.io/JamieCurnow/pen/KKPjraK
+    swatchStyle() {
+      const { tagColor, tagColorPickerOpen } = this
+      return {
+        backgroundColor: tagColor,
+        cursor: 'pointer',
+        height: '30px',
+        width: '30px',
+        borderRadius: tagColorPickerOpen ? '50%' : '4px',
+        transition: 'border-radius 200ms ease-in-out'
+      }
+    }
+  },
   methods: {
     ...mapActions(["addTag","markTagNew"]),
     openDialog({ tagType, tagQuery, tagStreams }) {
       this.tagType = tagType;
       this.tagQuery = tagQuery;
       this.tagStreams = tagStreams;
       this.tagName = "";
+      this.tagColor = "#81D4FA";
```

Comment:
we should somehow find a random color everytime the dialog is opened, otherwise we will end up with all tags having the same color